src/algebra/big_operators/basic.lean

```diff
@@ -913,41 +913,66 @@ lemma prod_range_one (f : ℕ → β) :
   ∏ k in range 1, f k = f 0 :=
 by { rw [range_one], apply @prod_singleton β ℕ 0 f }
 
-open multiset
+open list
 
-@[to_additive] lemma prod_multiset_map_count [decidable_eq α] (s : multiset α)
+@[to_additive] lemma prod_list_map_count [decidable_eq α] (l : list α)
   {M : Type*} [comm_monoid M] (f : α → M) :
-  (s.map f).prod = ∏ m in s.to_finset, (f m) ^ (s.count m) :=
+  (l.map f).prod = ∏ m in l.to_finset, (f m) ^ (l.count m) :=
 begin
-  induction s using multiset.induction_on with a s ih,
-  { simp only [prod_const_one, count_zero, prod_zero, pow_zero, multiset.map_zero] },
-  simp only [multiset.prod_cons, map_cons, to_finset_cons, ih],
+  induction l with a s IH, { simp only [map_nil, prod_nil, count_nil, pow_zero, prod_const_one] },
+  simp only [list.map, list.prod_cons, to_finset_cons, IH],
   by_cases has : a ∈ s.to_finset,
-  { rw [insert_eq_of_mem has, ← insert_erase has, prod_insert (not_mem_erase _ _),
-        prod_insert (not_mem_erase _ _), ← mul_assoc, count_cons_self, pow_succ],
-    congr' 1, refine prod_congr rfl (λ x hx, _),
+  { simp only [has, insert_eq_of_mem],
+    rw ← insert_erase has,
+    simp only [prod_insert (not_mem_erase _ _), ←mul_assoc, count_cons_self, pow_succ],
```

Comment:
Does `finset.mul_prod_erase` or similar help shorten this proof?

Reply:
I've investigated this and I can't see how to use it (or something similar) to shorten the proof.